web/src/client/components/session-view/mobile-input-overlay.ts

```diff
@@ -54,12 +74,45 @@ export class MobileInputOverlay extends LitElement {
 
   private handleMobileInputChange(e: Event) {
     const textarea = e.target as HTMLTextAreaElement;
+
+    // Skip processing if we're in the middle of IME composition
+    if (this.isComposing) {
+      return;
+    }
+
     this.mobileInputText = textarea.value;
     this.onTextChange?.(textarea.value);
     // Force update to ensure button states update
     this.requestUpdate();
   }
 
+  private handleCompositionStart = (_e: CompositionEvent) => {
```

Comment:
wonder if it's worth extracting that into a helper since it's quite a bit of code duplication.